quickwit-search/src/leaf.rs

```diff
@@ -267,12 +266,17 @@ pub async fn leaf_search(
                 Err(err) => Either::Right(err),
             });
 
+    // create a collector which merges responses into one
     let merge_collector = make_merge_collector(request);
-    let mut merged_search_response =
-        spawn_blocking(move || merge_collector.merge_fruits(split_search_responses))
-            .instrument(info_span!("merge_search_responses"))
-            .await
-            .context("Failed to merge split search responses.")??;
+
+    let mut merged_search_response = {
+        let span = info_span!("merge_search_responses");
+        let _enter = span.enter();
+
+        merge_collector
+            .merge_fruits(split_search_responses)
+            .context("Failed to merge split search responses.")?
+    };
```

Comment:
Spawning task into tokio runtime in this case might introduce a little overhead for scheduler. In this way, it might be better to run merging task right away.

Reply:
We avoid running any blocking operation that are not trivially taking less than 500 microseconds.

Reply:
Previous code looks like:
```
 let mut merged_search_response =
        spawn_blocking(move || merge_collector.merge_fruits(split_search_responses))
            .instrument(info_span!("merge_search_responses"))
            .await
            .context("Failed to merge split search responses.")??;
```
spawn task into runtime, let other/same thread run it. Current task is awaited for spawned task to be done. 

Hmm, I do not see the benefit here.

Reply:
spawn_blocking is different from spawn. It spawns a tasks a different thread pool than the tokio runtime thread pool.

The tokio runtime is meant to run a lot of IO operations and assumes cooperative tasks.
When we block it for say 1 second, io intensive operations will be blocked for no reason at all.

Reply:
Oh if the metrics show that merging is cpu-bound tasks and often takes long, I think you are right to spawn it into Tokio's blocking threads.

Will revert the changes. I just read the code and does not know the reason why doing this. Let me add comment also, in this way, other developer does not need to ask like me :heart:

Reply:
Sure. It is a very common pitfall when using tokio. I think we have a point about it in our code style.
https://github.com/quickwit-oss/quickwit/blob/main/CODE_STYLE.md